XPlat/Shopify/iOS/source/Shopify.iOS/ApiDefinition.cs

```diff
@@ -0,0 +1,1964 @@
+		NSDecimalNumber Price { get; set; }
+
+		[Export ("rate", ArgumentSemantic.Strong)]
+		NSDecimalNumber Rate { get; set; }
+
+		[Export ("title")]
+		string Title { get; set; }
+
+	}
+
+	[Category]
+	[BaseType (typeof(Theme), Name = "BUYTheme")]
+	interface ThemeExtensions
+	{
+		[Export ("backgroundColor")]
+		UIColor BackgroundColor ();
+
+		[Export ("selectedBackgroundColor")]
+		UIColor SelectedBackgroundColor ();
+
+		[Export ("separatorColor")]
+		UIColor SeparatorColor ();
+
+		[Export ("disclosureIndicatorColor")]
+		UIColor DisclosureIndicatorColor ();
+
+		[Export ("checkoutButtonTextColor")]
+		UIColor CheckoutButtonTextColor ();
+
+		[Export ("errorTintOverlayColor")]
+		UIColor ErrorTintOverlayColor ();
+
+		[Export ("navigationBarTitleColor")]
+		UIColor NavigationBarTitleColor ();
+
+		[Export ("navigationBarTitleVariantSelectionColor")]
+		UIColor NavigationBarTitleVariantSelectionColor ();
+
+		[Export ("productTitleColor")]
+		UIColor ProductTitleColor ();
+
+		[Export ("variantOptionNameTextColor")]
+		UIColor VariantOptionNameTextColor ();
+
+		[Export ("variantBreadcrumbsBackground")]
+		UIColor VariantBreadcrumbsBackground ();
+
+		[Export ("blurEffect")]
+		UIBlurEffect BlurEffect ();
+
+		[Export ("activityIndicatorViewStyle")]
+		UIActivityIndicatorViewStyle ActivityIndicatorViewStyle ();
+
+		[Export ("navigationBarStyle")]
+		UIBarStyle NavigationBarStyle ();
+
+		[Export ("paymentButtonStyle")]
+		PaymentButtonStyle PaymentButtonStyle ();
+
+	}
+
+	// @interface BUYVariantOptionView : UIView <BUYThemeable>
+	[BaseType (typeof(UIView), Name = "BUYVariantOptionView")]
+	interface VariantOptionView : Themeable
+	{
+		// -(void)setTextForOptionValue:(BUYOptionValue *)optionValue;
+		[Export ("setTextForOptionValue:")]
+		void SetTextForOptionValue (OptionValue optionValue);
+	}
+
+	interface IVariantSelectionDelegate
+	{
+
+	}
+
+	// @protocol BUYVariantSelectionDelegate <NSObject>
+	[Protocol, Model]
+	[BaseType (typeof(NSObject), Name = "BUYVariantSelectionDelegate")]
+	interface VariantSelectionDelegate
+	{
+		// @required -(void)variantSelectionController:(BUYVariantSelectionViewController *)controller didSelectVariant:(BUYProductVariant *)variant;
+		[Abstract]
+		[Export ("variantSelectionController:didSelectVariant:")]
+		void VariantSelectionController (VariantSelectionViewController controller, ProductVariant variant);
+
+		// @required -(void)variantSelectionControllerDidCancelVariantSelection:(BUYVariantSelectionViewController *)controller atOptionIndex:(NSUInteger)optionIndex;
+		[Abstract]
+		[Export ("variantSelectionControllerDidCancelVariantSelection:atOptionIndex:")]
+		void VariantSelectionControllerDidCancelVariantSelection (VariantSelectionViewController controller, nuint optionIndex);
+	}
+
+	// @interface BUYVariantSelectionViewController : UIViewController
+	[BaseType (typeof(UIViewController), Name = "BUYVariantSelectionViewController")]
+	interface VariantSelectionViewController
```

Comment:
Checking with header, I would suggest to add [DisableDefaultCtor] attribute because Product parameter cannot be set after instantiating.